Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -880,27 +907,18 @@ function Set-AvsVMStoragePolicy {
     } 
 
     $ProtectedVMs = Get-ProtectedVMs 
-    if ($ProtectedVMs.Name.Contains($VMName)) {
-        Write-Error "Access denied to this VM." -ErrorAction Stop
-    }
-    $VM = Get-VM $VMName
-    if ($null -eq $VM) {
-        Write-Error "Was not able to set the storage policy on the VM. Could not find VM with the name: $VMName" -ErrorAction Stop
-    }
-    if (-not $(Get-SpbmEntityConfiguration $VM).StoragePolicy -in $VSANStoragePolicies) {
-        Write-Error "Modifying Storage policy on this VM is not supported" -ErrorAction Stop
-    }
-    Write-Host "Setting VM $VMName storage policy to $StoragePolicyName..."
-    try {
-        Set-VM -VM $VM -StoragePolicy $StoragePolicy -ErrorAction Stop -Confirm:$false
-    }
-    catch [VMware.VimAutomation.ViCore.Types.V1.ErrorHandling.InvalidVmConfig] {
-        Write-Error "The selected storage policy $($StoragePolicy.Name) is not compatible with this VM. You may need more hosts: $($PSItem.Exception.Message)" -ErrorAction Stop
-    }
-    catch {
-        Write-Error "Was not able to set the storage policy on the VM: $($PSItem.Exception.Message)" -ErrorAction Stop
-    }
-    Write-Output "Successfully set the storage policy on VM $VMName to $StoragePolicyName"
+    $VMList = Get-VM $VMName | Where-Object {-not $_.Name -in $ProtectedVMs.Name}
+
+    if ($null -eq $VMList) {
+        Write-Error "Was not able to set the storage policy on the VM. Could not find VM(s) with the name: $VMName" -ErrorAction Stop
+    } elseif ($VMList.count -eq 1) {
+        $VM = $VMList[0]
+        Set-StoragePolicyOnVM -VM $VM -VSANStoragePolicies $VSANStoragePolicies -StoragePolicy $StoragePolicy -ErrorAction Stop
```

Comment:
What happens if the -ErrorAction is Continue, in the case of a single VM in the VMList?

Reply:
It would print the error out, but continue and not terminate. In other words, it would show up in that Error pane but the status of the execution would still say "Succeeded" because an exception wasn't thrown. In this case, if the user is trying to set a specific VM's StoragePolicy and it fails, it'll fail the whole execution and say it "Failed" and print this line as the reason. If it were set to Continue, this line would print in the Error tab but then it would still successfully complete since this is more synonymous to a tracer.Errorf line.